patches/random_costs.cpp

```diff
@@ -0,0 +1,284 @@
+#include <string>
+#include <vector>
+#include <algorithm>
+#include <random>
+#include "genie/dat/DatFile.h"
+#include "ids.h"
+
+
+typedef genie::ResourceUsage<int16_t, int16_t, int16_t> ResourceCost;
+typedef genie::ResourceUsage<int16_t, int16_t, int8_t> ResearchResourceCost;
+
+bool isNaturalResourceCost(const ResourceCost &cost) {
+    return cost.Type != -1 && cost.Type < 4 && cost.Amount > 0;
+}
+
+bool isNaturalResearchResourceCost(const ResearchResourceCost &cost) {
+    return cost.Type != -1 && cost.Type < 4 && cost.Amount > 0;
+}
+
+bool hasNaturalResourceCost(const genie::Unit &unit) {
+	std::vector<ResourceCost> costs = unit.Creatable.ResourceCosts;
```

Comment:
This looks like there are a few issues with spaces and tabs in this file :thinking:

Reply:
whoops, my bad